### PR TITLE
Xcode markup documentation

### PR DESCRIPTION
**Pull request by tomj**
So @eimantas has cranked out an _insane_ amount of documentation over the past couple of days.  I've been proofreading and cleaning up the commit history a little but I think it looks pretty good.

I've added what has been done for `Scheduler.swift` to open up the discussion on this PR.  Given the volume of changes which have been made (and still need to be proofread/polished) we thought it would be good to hear what everyone thought before making a whopper of a PR.  Any feedback appreciated 😄 🎉

### CODE REVIEW

**Comment by tomj** [comment]
(fyi the remaining changes which @eimantas made are on [my fork](https://github.com/tomj/ReactiveCocoa/commits/master))

cc #2982

**Comment by mdiep** [comment]
Awesome! I'm traveling back from WWDC today, but will try to take a look sometime in the next few days.

**Comment by mdiep** [comment]
This looks great to me!

**Comment by eimantas** [comment]
Should we proceed with PR per file or one PR for all files?

**Comment by mdiep** [comment]
My vote would be for one PR.

**Comment by tomj** [comment]
Nice!  Thanks @mdiep 😄  I'll go through and do some more proofreading with @eimantas and try and get the rest pushed up to this PR asap.

File: ReactiveCocoa/Swift/Event.swift [hunk 1]
```diff
@@ -56,6 +64,11 @@ public enum Event<Value, Error: ErrorType> {
 	}
 
 	/// Lifts the given function over the event's error.
+	///
+	/// - parameters:
+    ///   - f: A closure that accepts an error and returns
```
Comment (eimantas) [inline ReactiveCocoa/Swift/Event.swift]: Misalignment here

File: ReactiveCocoa/Swift/Observer.swift [hunk 2]
```diff
@@ -31,10 +31,21 @@ public struct Observer<Value, Error: ErrorType> {
 
 	public let action: Action
 
+	/// Initializer that accepts a closure that processes the event
+	///
+	/// - parameters:
+    ///   - action: A closure of type `Event<Value, Error: ErrorType> -> Void`
```
Comment (eimantas) [inline ReactiveCocoa/Swift/Observer.swift]: Ditto for misalignment

File: ReactiveCocoa/Swift/Observer.swift [hunk 3]
```diff
@@ -56,11 +67,17 @@ public struct Observer<Value, Error: ErrorType> {
 
 extension Observer: ObserverType {
 	/// Puts a `Next` event into the given observer.
+	///
+	/// - parameters:
+    ///   - value: a value sent with the `Next` event
```
Comment (eimantas) [inline ReactiveCocoa/Swift/Observer.swift]: Aaand here.

Comment (tomj) [inline ReactiveCocoa/Swift/Observer.swift]: Ha! 😄   So I'm cherry picking the commit you cooked up with the indentation changes which will fix this up.  At the same time I need to investigate why my Xc is being so weird with tabs and spaces, afaiknew it normally did conversions in place.

Comment (tomj) [inline ReactiveCocoa/Swift/Observer.swift]: Aaaand I just realised that this project is tabs not spaces.  My bad, I'll tidy this up on a branch on my fork and will push the rest of these commits up to here when they're all done 😄 

https://www.youtube.com/watch?v=SsoOG6ZeyUI

**Comment by mdiep** [comment]
Is this ready for review??

**Comment by tomj** [comment]
@mdiep ready from my perspective.  We still need to add the docs for `Signal` and `SignalProducer`.  Let me know what you think about adding these and the concluding with those 2 classes.

**Comment by mdiep** [comment]
I'd love to have the `Signal` and `SignalProducer` docs converted here as well. I'll go ahead and review what's been done!

**Comment by tomj** [comment]
Completely understand - I kinda felt the same way too 😃.  Will rebase on `master` this weekend and try and crank out the remainder where possible

File: ReactiveCocoa/Swift/Disposable.swift [hunk 4]
```diff
@@ -234,20 +264,30 @@ public final class SerialDisposable: Disposable {
...
+/// ````
+///  disposable += producer
+///      .filter { ... }
+///      .map    { ... }
+///      .start(observer)
+/// ````
 ///
+/// - parameters:
+///   - lhs: Disposable to add to.
+///   - rhs: Disposable to add.
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Disposable.swift]: This and the variant below are missing the `returns`.

File: ReactiveCocoa/Swift/Event.swift [hunk 5]
```diff
@@ -38,7 +40,13 @@ public enum Event<Value, Error: ErrorType> {
...
 	}
 
-	/// Lifts the given function over the event's value.
+	/// Lifts the given closure over the event's value.
+	///
+	/// - parameters:
+	///   - f: A closure that accepts a value and returns
+	///        a new value
+	///
+	/// - important: The closure is called only on `Next` type events.
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Event.swift]: This and `mapError` below don't specify `returns:`.

File: ReactiveCocoa/Swift/Flatten.swift [hunk 6]
```diff
@@ -59,11 +59,11 @@ extension SignalType where Value: SignalProducerType, Error == NoError {
 	/// Flattens the inner producers sent upon `signal` (into a single signal of
 	/// values), according to the semantics of the given strategy.
 	///
-	/// If an active inner producer fails, the returned signal will forward that
-	/// failure immediately.
-	///
-	/// `Interrupted` events on inner producers will be treated like `Completed`
-	/// events on inner producers.
+	/// - parameter strategy: Strategy used when flattening signals.
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Flatten.swift]: This breaks the rule about using `- parameters:` in the documentation documentation. ☺️

File: ReactiveCocoa/Swift/Observer.swift [hunk 7]
```diff
@@ -29,12 +29,27 @@ public protocol ObserverType {
...
 	}
 
+	/// Initializer that accepts closures for different event types.
+	///
+	/// - parameters:
+	///   - failed: Optional closure that accepts an `Error` parameter when a
+	///             `Failed` event is observed.
+	///	  - completed: Optional closure executed when a `Completed` event is
+	///                observed.
+	///	  - interruped: Optional closure executed when an `Interrupted` event is
```
Comment (mdiep) [inline ReactiveCocoa/Swift/Observer.swift]: These seem mis-indented.

**Comment by mdiep** [comment]
This is one of my favorite PRs of all time—especially `DocumentingCode.md`. 😍 

I spotted a few minor issues. I'd love to get those fixed up, the rest of it documented, and then get this merged. We're gearing up for a Swift 2.3 release, and I'd like to include this. Plus the ongoing RAC 5 work will introduce more and more conflicts, so it'd be good to get this merged ASAP.

Do you know when you'll have time to finish this?

**Comment by tomj** [comment]
it's almost all @eimantas - I just did some proofreading 😄

I just rebased on the current `master` (which was luckily conflict free, I really should have kept it a bit more fresh), I'll aim to clean up those small issues tonight and jump on `Signal` and `SignalProducer` before Friday (I had less time for this over the weekend than I was hoping).

**Comment by eimantas** [comment]
To my knowledge all code is marked up. So it's reading time!

Since `DocumentingCode.md` was created midway the markup process I'm pretty sure there will be incompliances with the guidelines. If it's not too much - this may as well be merged and fixes can be applied in master after last 4.x release has been done.

File: ReactiveCocoa/Swift/Observer.swift [hunk 8]
```diff
@@ -40,16 +40,16 @@ public struct Observer<Value, Error: ErrorType> {
 		self.action = action
 	}
 
-	/// Initializer that accepts closures for different event types
+	/// Initializer that accepts closures for different event types.
```
Comment (andersio) [inline ReactiveCocoa/Swift/Observer.swift]: "An initializer", perhaps?

File: ReactiveCocoa/Swift/Observer.swift [hunk 9]
```diff
@@ -40,16 +40,16 @@ public struct Observer<Value, Error: ErrorType> {
...
 	///
 	/// - parameters:
-	///   - failed: Optional closure that accepts `Error` parameter when
+	///   - failed: Optional closure that accepts an `Error` parameter when a
 	///             `Failed` event is observed.
-	///	  - completed: Optional closure executed when `Completed` event is
```
Comment (andersio) [inline ReactiveCocoa/Swift/Observer.swift]: #TabsVsSpaces

**Comment by andersio** [comment]
Comments in `Observer.swift` like these
```
	/// Puts a `Failed` event into the given observer.
```
do not seem quite right. Instead of `the given observer`, ``` `self` ``` sounds more appropriate.

**Comment by mdiep** [comment]
This is fantastic. Thank you so much!

It looks good to me, but if I missed anything we can fix it up later.

**Pull request merged**